docker/src/main/java/org/jclouds/docker/config/DockerHttpApiModule.java

```diff
@@ -48,8 +65,45 @@ protected void bindErrorHandlers() {
    @Override
    protected void configure() {
       super.configure();
-      install(new OkHttpCommandExecutorServiceModule());
-      bind(OkHttpClientSupplier.class).to(DockerOkHttpClientSupplier.class);
+      install(new SSLModule());
+      bind(HttpCommandExecutorService.class).to(OkHttpCommandExecutorService.class).in(Scopes.SINGLETON);
+      bind(OkHttpClient.class).toProvider(DockerOkHttpClientProvider.class).in(Scopes.SINGLETON);
+  }
+
+   private static final class DockerOkHttpClientProvider implements Provider<OkHttpClient> {
+      private final HttpUtils utils;
+      private final Supplier<SSLContext> sslContextWithKeysSupplier;
+      private final OkHttpClientSupplier clientSupplier;
+      private final HostnameVerifier hostnameVerifier;
+
+      @Inject
+      DockerOkHttpClientProvider(OkHttpClientSupplier clientSupplier, HttpUtils utils, SSLContextWithKeysSupplier sslContextWithKeysSupplier, @Named("untrusted") HostnameVerifier hostnameVerifier) {
+         this.clientSupplier = clientSupplier;
+         this.utils = utils;
+         this.sslContextWithKeysSupplier = sslContextWithKeysSupplier;
+         this.hostnameVerifier = hostnameVerifier;
+      }
+
+      @Override
+      public OkHttpClient get() {
+         OkHttpClient client = clientSupplier.get();
+         client.setConnectTimeout(utils.getConnectionTimeout(), TimeUnit.MILLISECONDS);
+         client.setReadTimeout(utils.getSocketOpenTimeout(), TimeUnit.MILLISECONDS);
+         client.setFollowRedirects(false);
+         ConnectionSpec tlsSpec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                 .tlsVersions(TlsVersion.TLS_1_0, TlsVersion.TLS_1_1, TlsVersion.TLS_1_2)
+                 .build();
+         ConnectionSpec cleartextSpec = new ConnectionSpec.Builder(ConnectionSpec.CLEARTEXT)
+                 .build();
+         client.setConnectionSpecs(ImmutableList.of(tlsSpec, cleartextSpec));
+         client.setSslSocketFactory(sslContextWithKeysSupplier.get().getSocketFactory());
+
+         if (utils.relaxHostname()) {
+            client.setHostnameVerifier(hostnameVerifier);
+         }
+         return client;
+      }
```

Comment:
Feedback is always welcome, so +1 to log what is going on. There should no ned to provide a custom unstrusted ssl context for Docker. The logging can happen in the SSLContextWithKeysSupplier get() method, saying that since the trust all certs property has been enabled its configuration will be ignored. At that point you should have everything you need to build the log (and the injector will already have been created so you can properly use an injected logger).